### PR TITLE
add tags to host and volume resource

### PR DESCRIPTION
**Pull request by AnuBose**
add tags to host & volume resources. Able to create and view the tags.

### CODE REVIEW

**Comment by neilgierman** [comment]
Are there changes to https://github.com/hpe-hcss/quake-client needed also?

File: internal/resources/utils.go [hunk 1]
```diff
@@ -152,3 +152,19 @@ func wrapResourceError(err *error, msg string) {
 
 	*err = fmt.Errorf("%s %w", msg, *err)
 }
+
+// converMap returns map of string key to string value.
+func converMap(in map[string]interface{}) map[string]string {
```
Comment (ron-saito) [inline internal/resources/utils.go:157]:
```suggestion
func convertMap(in map[string]interface{}) map[string]string {
```

File: internal/resources/utils.go [hunk 2]
```diff
@@ -152,3 +152,19 @@ func wrapResourceError(err *error, msg string) {
 
 	*err = fmt.Errorf("%s %w", msg, *err)
 }
+
+// converMap returns map of string key to string value.
```
Comment (ron-saito) [inline internal/resources/utils.go:156]:
```suggestion
// convertMap returns map of string key to string value.
```

File: internal/resources/resource_volume.go [hunk 3]
```diff
@@ -252,6 +263,12 @@ func resourceMetalVolumeRead(d *schema.ResourceData, meta interface{}) (err erro
 	d.Set(vState, volume.State)
 	d.Set(vStatus, volume.Status)
 
+	tags := make(map[string]string)
```
Comment (ron-saito) [inline internal/resources/resource_volume.go:266]:
```suggestion
	tags := make(map[string]string, len(tags))
```

File: internal/resources/resource_host.go [hunk 4]
```diff
@@ -474,6 +486,16 @@ func resourceMetalHostRead(d *schema.ResourceData, meta interface{}) (err error)
 		return err
 	}
 
+	tags := make(map[string]string)
```
Comment (ron-saito) [inline internal/resources/resource_host.go:489]:
```suggestion
	tags := make(map[string]string, len(tags))
```

File: internal/resources/resource_host.go [hunk 5]
```diff
@@ -396,6 +402,12 @@ func resourceMetalHostCreate(d *schema.ResourceData, meta interface{}) (err erro
 		host.PreAllocatedIPs = convertStringArr(ips)
 	}
 
+	host.Labels = make(map[string]string)
```
Comment (ron-saito) [inline internal/resources/resource_host.go:405]:
```suggestion
	host.Labels = make(map[string]string, len(host.Labels))
```

**Comment by neilgierman** [comment]
I jut published a release for terraform client. Please update go.mod to pull v1.4.5.

**Comment by neilgierman** [comment]
> I jut published a release for terraform client. Please update go.mod to pull v1.4.5.

Nevermind. Doing the update in https://github.com/HewlettPackard/hpegl-metal-terraform-resources/pull/60

File: examples/resources/hpegl_metal_host/resource.tf [hunk 6]
```diff
@@ -70,4 +70,5 @@ resource "hpegl_metal_host" "terra_host_new_ssh" {
   network_route = "Public"
   location      = var.location
   description   = "Hello from Terraform"
+  labels        = {"ServiceType" = "BMaaS"}
```
Comment (neilgierman) [inline examples/resources/hpegl_metal_host/resource.tf:73]:
```suggestion
  labels        = { "ServiceType" = "BMaaS" }
```

**Review by ron-saito — APPROVED** [review]

**Pull request merged**